beanie/odm/utils/init.py

```diff
@@ -258,19 +281,31 @@ async def init_indexes(cls, allow_index_dropping: bool = False):
         new_indexes = ["_id_"]
 
         # Indexed field wrapped with Indexed()
-        found_indexes = [
-            IndexModel(
-                [
-                    (
-                        fvalue.alias,
-                        fvalue.type_._indexed[0],
+        found_indexes = []
+        for fvalue in cls.__fields__.values():
+            if getattr(fvalue.type_, "_indexed", False):
+                kwargs = fvalue.type_._indexed[1]
+
+                # if indexed field is Optional, index creation should be filtered
+                print(
```

Comment:
Please, remove this

Reply:
Yes, it was accidentally appeared here, this is another feature implementation.